tests/browser/test/clipboard_test.mjs

```diff
@@ -0,0 +1,611 @@
+    chai.assert.equal(
+      focusedBlockType,
+      'logic_boolean',
+      'Newly pasted block should be selected',
+    );
+  });
+
+  test('Copying a block also copies and pastes its children', async function () {
+    await loadStartBlocks(this.browser);
+    // Select and copy the "if/else" block which has children
+    await focusOnBlock(this.browser, 'controls_if_1');
+    await this.browser.pause(PAUSE_TIME);
+
+    await this.browser.keys([Key.Ctrl, 'c']);
+    await this.browser.pause(PAUSE_TIME);
+
+    // Check how many blocks there are before pasting
+    const allBlocksBeforePaste = await getAllBlocks(this.browser);
+
+    // Paste the block while still in the main workspace
+    await this.browser.keys([Key.Ctrl, 'v']);
+    await this.browser.pause(PAUSE_TIME);
+
+    // Check result
+    const allBlocksAfterPaste = await getAllBlocks(this.browser);
+    chai.assert.equal(
+      allBlocksAfterPaste.length,
+      allBlocksBeforePaste.length + 4,
+      'Expected there to be four additional blocks after paste',
+    );
+  });
+
+  test('Paste shadow block to/from main workspace', async function () {
+    await loadStartBlocks(this.browser);
+    // Select and copy the shadow number block
+    await focusOnBlock(this.browser, 'math_number_shadow_1');
+    await this.browser.pause(PAUSE_TIME);
+
+    await this.browser.keys([Key.Ctrl, 'c']);
+    await this.browser.pause(PAUSE_TIME);
+
+    // Check how many blocks there are before pasting
+    const allBlocksBeforePaste = await getAllBlocks(this.browser);
+
+    // Paste the block while still in the main workspace
+    await this.browser.keys([Key.Ctrl, 'v']);
+    await this.browser.pause(PAUSE_TIME);
+
+    // Check result
+    const allBlocksAfterPaste = await getAllBlocks(this.browser);
+    chai.assert.equal(
+      allBlocksAfterPaste.length,
+      allBlocksBeforePaste.length + 1,
+      'Expected there to be one additional block after paste',
+    );
+    const focusedBlockId = await getSelectedBlockId(this.browser);
+    chai.assert.notEqual(
+      focusedBlockId,
+      'math_number_shadow_1',
+      'Newly pasted block should be selected',
+    );
+    const focusedBlockType = await getSelectedBlockType(this.browser);
+    chai.assert.equal(
+      focusedBlockType,
+      'math_number',
+      'Newly pasted block should be selected',
+    );
+    const focusedBlockIsShadow = await this.browser.execute(() => {
+      return Blockly.common.getSelected().isShadow();
+    });
+    chai.assert.isFalse(
+      focusedBlockIsShadow,
+      'Expected the pasted version of the block to not be a shadow block',
+    );
+  });
+
+  test('Copy block from flyout, paste to main workspace', async function () {
+    // Open flyout
+    await getCategory(this.browser, 'Logic').then((category) =>
+      category.click(),
+    );
+
+    // Focus on first block in flyout
+    await this.browser.execute(() => {
+      const ws = Blockly.getMainWorkspace().getFlyout().getWorkspace();
```

Comment:
Should this get pulled out into a helper method in test_setup since you use it a few times?